scss/components/_ratio-box.scss

```diff
@@ -0,0 +1,69 @@
+
+.ratio-box--16-9 {
+  position: relative;
+
+  &::before {
+    content: '';
+    padding-top: 56.25%;
+    display: block;
+  }
+
+  > * {
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    object-fit: cover;
+  }
+}
+
+.ratio-box--4-3 {
+  position: relative;
+
+  &::before {
+    content: '';
+    padding-top: 75%;
+    display: block;
+  }
+
+  > * {
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    object-fit: cover;
+  }
+}
+
+.ratio-box--1-1 {
+  position: relative;
+
+  &::before {
+    content: '';
+    padding-top: 100%;
+    display: block;
+  }
+
+  > * {
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    object-fit: cover;
+  }
+}
+
```

Comment:
with the #2 assignment: please add here a reusable class, lets say `responsive-ratio-box`, it should size its contents to an aspect ratio of 4:3 whenever the screen is in "portrait" view and to an aspect ratio of 16:9 whenever the screen is in "landscape mode"